### PR TITLE
Do not create packagejson imports and auto import provider in partial semantic server mode

### PR DESCRIPTION
**Pull request by sheetalkamat**
Found this while looking into #39656 where it tried to read package.json file

### CODE REVIEW

File: src/server/project.ts [hunk 1]
```diff
@@ -1622,6 +1622,7 @@ namespace ts.server {
 
         /*@internal*/
         getPackageJsonsVisibleToFile(fileName: string, rootDir?: string): readonly PackageJsonInfo[] {
+            if (this.projectService.serverMode !== LanguageServiceMode.Semantic) return emptyArray;
```
Comment (DanielRosenwasser) [inline src/server/project.ts:1625]: Slightly afraid of all the places that are going to need to be patched up in the future to check like this, but I guess that's programming

**Review by DanielRosenwasser — APPROVED** [review]
Get @andrewbranch to review but looks good to me.

File: src/testRunner/unittests/tsserver/partialSemanticServer.ts [hunk 2]
```diff
@@ -172,5 +172,36 @@ function fooB() { }`
             openFilesForSession([file2], session);
             checkProjectActualFiles(project, [libFile.path, file2.path, file3.path]);
         });
+
+        it("should not create autoImportProvider or handle package jsons", () => {
```
Comment (andrewbranch) [inline src/testRunner/unittests/tsserver/partialSemanticServer.ts:176]:
```suggestion
        it("should not create autoImportProvider or handle package jsons in PartialSemantic mode", () => {
```

**Review by andrewbranch — APPROVED** [review]

File: src/testRunner/unittests/tsserver/partialSemanticServer.ts (hunk 2 above)
Comment (sheetalkamat) [reply src/testRunner/unittests/tsserver/partialSemanticServer.ts:176]: Its already in describe block for with `unittests:: tsserver:: Semantic operations on PartialSemantic server`

Comment (andrewbranch) [reply src/testRunner/unittests/tsserver/partialSemanticServer.ts:176]: Ah, missed what file this was.

**Pull request merged**